engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/deployer/BpmnDeployer.java

```diff
@@ -293,22 +293,11 @@ protected List<EventSubscriptionEntity> getPreviousSubscriptionEvents(ProcessDef
   }
 
   protected List<EventSubscriptionEntity> getOrphanSubscriptionEvents(ProcessDefinitionEntity processDefinition) {
-    Map<String, EventSubscriptionDeclaration> eventSubscriptionDeclarations = processDefinition.getProperties().get(BpmnProperties.EVENT_SUBSCRIPTION_DECLARATIONS);
-
-    // if finding subscriptions by name and tenant ID impacts performance, we could think about another way of filtering
-    // (for example doing "configuration" LIKE "processDefinition.key%")
-    if (!eventSubscriptionDeclarations.isEmpty()) {
-      return eventSubscriptionDeclarations.values()
-          .stream()
-          .filter(EventSubscriptionDeclaration::hasEventName)
-          .flatMap(declaration -> getEventSubscriptionManager().findEventSubscriptionsByNameAndTenantId(
-                  declaration.getEventType(), declaration.getUnresolvedEventName(), processDefinition.getTenantId())
-              .stream())
-          .filter(this::isOrphan)
-          .collect(Collectors.toList());
-    }
-
-    return Collections.emptyList();
+    String configurationLike = processDefinition.getKey() + ":%:%";
```

Comment:
💭🙃 FYI: I discussed this part with Yana. We were unsure if there could be edge cases where the pattern yields unwanted results. Example: The pattern would match two process definitions named `myProcDef:1:2` and `myProc:Def:1:2`. However, colons are not allowed as part of the process definition id. It should be safe to use.

Reply:
Yes I thought about this. We're also doing another check after to make sure the subscription is orphan before deleting.